instrumentation/kamon-armeria/src/test/scala/kamon/instrumentation/armeria/client/ArmeriaHttpClientTracingSpec.scala

```diff
@@ -0,0 +1,231 @@
+package kamon.instrumentation.armeria.client
+
+import com.linecorp.armeria.client.{ClientFactory, Clients, WebClient}
+import com.linecorp.armeria.common.{HttpMethod, HttpRequest, RequestHeaders}
+import kamon.Kamon
+import kamon.context.Context
+import kamon.tag.Lookups.{plain, plainBoolean, plainLong}
+import kamon.testkit.{InitAndStopKamonAfterAll, TestSpanReporter}
+import kamon.trace.Span
+import kamon.trace.SpanPropagation.B3
+import org.scalatest.concurrent.Eventually
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.time.SpanSugar.convertIntToGrainOfTime
+import org.scalatest.wordspec.AnyWordSpec
+import org.scalatest.{BeforeAndAfterAll, OptionValues}
+import utils.ArmeriaServerSupport.startArmeriaServer
+import utils.TestSupport.getResponseHeaders
+
+class ArmeriaHttpClientTracingSpec extends AnyWordSpec
+  with Matchers
+  with BeforeAndAfterAll
+  with InitAndStopKamonAfterAll
+  with Eventually
+  with TestSpanReporter
+  with OptionValues {
+
+  val customTag = "requestId"
+  val customHeaderName = "X-Request-Id"
+
+  val interface = "127.0.0.1"
+  val httpPort = 8080
+
+  private val httpServer = startArmeriaServer(httpPort)
+
+  "The Armeria http client tracing instrumentation" should {
+
+    "propagate the current context and generate a span around an async request" in {
+      val path = "/users"
+      val url = s"http://$interface:$httpPort"
+
+      val okSpan = Kamon.spanBuilder("ok-async-operation-span").start()
+      val client = Clients.builder(url).build(classOf[WebClient])
+      val request = HttpRequest.of(RequestHeaders.of(HttpMethod.GET, path))
+
+      val response = Kamon.runWithContext(Context.of(Span.Key, okSpan)) {
+        client.execute(request).aggregate().get()
+      }
+
+      val span = eventually(timeout(3 seconds)) {
+        val span = testSpanReporter().nextSpan().value
+
+        span.operationName shouldBe path
```

Comment:
I was debugging this and found that the operation name is set to the request path, which can have cardinality issues. I think when we remove the operation name generator things will be better since the requests will have the same default names as all other instrumentations.

Reply:
Seems like a blocker

Reply:
Yep, I think that, if the configuration is not properly provided, we could have cardinality issues because of:
- The current implementation of `KamonArmeriaHttpOperationNameGenerator`
- The fact that we don't have a method like `patternString()` in the client side